From 5347d7cd020a935cb0748adbdee2cf532c53c921 Mon Sep 17 00:00:00 2001
From: Illia Vysochyn <ivysochyn@internships.antmicro.com>
Date: Wed, 6 Apr 2022 14:35:56 +0200
Subject: [PATCH] Removes building of extensions

---
 setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.py b/setup.py
index 1979bac9..9101f5a1 100644
--- a/setup.py
+++ b/setup.py
@@ -277,7 +277,6 @@ cmdclass = {
     'create_version': create_version,
     'build_py': build_py,
     'develop': develop,
-    'build_ext': build_ext,
     'typecheck': mypy_type_check,
 }
 
-- 
2.35.1

