From b2ee6c8e449a646d1edabfa4d95abcf1a4b74f5e Mon Sep 17 00:00:00 2001
From: Illia Vysochyn <ivysochyn@internships.antmicro.com>
Date: Wed, 6 Apr 2022 11:53:24 +0200
Subject: [PATCH] CMake build directory specified by enviroment variable

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 946dd552..1979bac9 100644
--- a/setup.py
+++ b/setup.py
@@ -23,7 +23,7 @@ import multiprocessing
 TOP_DIR = os.path.realpath(os.path.dirname(__file__))
 SRC_DIR = os.path.join(TOP_DIR, 'onnx')
 TP_DIR = os.path.join(TOP_DIR, 'third_party')
-CMAKE_BUILD_DIR = os.path.join(TOP_DIR, '.setuptools-cmake-build')
+CMAKE_BUILD_DIR = os.getenv('CMAKE_BUILD_DIR')
 PACKAGE_NAME = 'onnx'
 
 WINDOWS = (os.name == 'nt')
-- 
2.35.1

