From 6ff6fa6d32aefa117b2bb41efc326fbd202b1dbf Mon Sep 17 00:00:00 2001
From: Illia Vysochyn <ivysochyn@internships.antmicro.com>
Date: Tue, 19 Apr 2022 17:10:18 +0200
Subject: [PATCH] Adds a builddir to look for libraries

---
 python/setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/python/setup.py b/python/setup.py
index 7ff897085..f31216227 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -47,7 +47,7 @@ def get_lib_path():
     exec(compile(open(libinfo_py, "rb").read(), libinfo_py, "exec"), libinfo, libinfo)
     version = libinfo["__version__"]
     if not CONDA_BUILD:
-        lib_path = libinfo["find_lib_path"]()
+        lib_path = libinfo["find_lib_path"](search_path=os.getenv("BUILDDIR"))
         libs = [lib_path[0]]
         if "runtime" not in libs[0]:
             for name in lib_path[1:]:
-- 
2.35.1

